pkg/apis/provisioning/v1alpha5/provisioner_validation.go

```diff
@@ -133,29 +133,22 @@ func (c *Constraints) validateTaints() (errs *apis.FieldError) {
 	return errs
 }
 
+// This function is used by the provisioner validation webhook to verify the provisioner requirements.
+// When this function is called, the provisioner's requirments do not include the requirements from labels.
+// Provisioner requirements only support well known labels.
 func (c *Constraints) validateRequirements() (errs *apis.FieldError) {
-	for i, requirement := range c.Requirements {
-		if err := validateRequirement(requirement); err != nil {
-			errs = errs.Also(apis.ErrInvalidArrayValue(err, "requirements", i))
+	for _, requirement := range c.Requirements.Requirements {
+		// Ensure requirements are well known
+		if !WellKnownLabels.Has(requirement.Key) {
```

Comment:
To support `IgnoredLabels` I think you just need another conditional here to allow IgnoredLables that are not "well-known".  IgnoredLabels are just labels that should not affect scheduling. For example, the in-tree EBS driver attached a region label to pods that request an EBS volume. However, Karpenter doesn't support cross-region provisioning, so we can safely ignore the region label.